src/g1.rs

```diff
@@ -399,15 +399,14 @@ impl G1Affine {
     /// exists within the $q$-order subgroup $\mathbb{G}_1$. This should always return true
     /// unless an "unchecked" API was used.
     pub fn is_torsion_free(&self) -> Choice {
-        const FQ_MODULUS_BYTES: [u8; 32] = [
-            1, 0, 0, 0, 255, 255, 255, 255, 254, 91, 254, 255, 2, 164, 189, 83, 5, 216, 161, 9, 8,
-            216, 57, 51, 72, 125, 157, 41, 83, 167, 237, 115,
-        ];
+        // Algorithm from Section 6 of https://eprint.iacr.org/2021/1130
+        // Updated proof of correctness in https://eprint.iacr.org/2022/352
+        //
+        // Check that endomorphism_p(P) == -[x^2] P
 
-        // Clear the r-torsion from the point and check if it is the identity
-        G1Projective::from(*self)
-            .multiply(&FQ_MODULUS_BYTES)
-            .is_identity()
+        let minus_x_squared_times_p = G1Projective::from(self).mul_by_x().mul_by_x().neg();
+        let endomorphism_p = endomorphism(self);
+        G1Affine::from(minus_x_squared_times_p).ct_eq(&endomorphism_p)
```

Comment:
This requires an inversion and two multiplications (to convert from projective to affine). Instead, use:
```suggestion
        minus_x_squared_times_p.ct_eq(&G1Projective::from(endomorphism_p))
```
which only requires four multiplications.

Performance improvements as a result of this optimisation (benchmarked on a MacBook M1):
```
G1Affine subgroup check time:   [71.103 us 71.137 us 71.177 us]                                    
                        change: [-28.345% -28.163% -28.009%] (p = 0.00 < 0.05)
                        Performance has improved.
Found 15 outliers among 100 measurements (15.00%)
  4 (4.00%) high mild
  11 (11.00%) high severe

G1Affine deserialize compressed point                                                                            
                        time:   [99.106 us 99.164 us 99.242 us]
                        change: [-22.051% -21.869% -21.665%] (p = 0.00 < 0.05)
                        Performance has improved.
Found 13 outliers among 100 measurements (13.00%)
  2 (2.00%) high mild
  11 (11.00%) high severe

G1Affine deserialize uncompressed point                                                                            
                        time:   [71.409 us 71.454 us 71.510 us]
                        change: [-28.092% -27.955% -27.820%] (p = 0.00 < 0.05)
                        Performance has improved.
Found 14 outliers among 100 measurements (14.00%)
  3 (3.00%) high mild
  11 (11.00%) high severe
```